common/db_test.go

```diff
@@ -656,6 +659,83 @@ func TestLoadWinningTickets_GivenTwoSessionsWithTickets_ReturnsAllTickets(t *tes
 	assert.Equal(recipientRand1, recipientRands[1])
 }
 
+func TestInsertMiniHeader_ReturnsFindLatestMiniHeader(t *testing.T) {
+	dbh, dbraw, err := TempDB(t)
+	defer dbh.Close()
+	defer dbraw.Close()
+	assert := assert.New(t)
+	require := require.New(t)
+	require.Nil(err)
+
+	h0 := defaultMiniHeader(t)
+	err = dbh.InsertMiniHeader(h0)
+	require.Nil(err)
+
+	h0db, err := dbh.FindLatestMiniHeader()
```

Comment:
I think its also worth checking that the `MiniHeader` returned by `FindLatestMiniHeader()` or `FindAllMiniHeadersSortedByNumber()` can contain multiple logs (at the moment looks like we're always inserting `defaultMiniHeader()` which only has a single log).

Reply:
Fixed in ced063e by adding a second log for uneven indices in tests and asserting the length of the logs contained in the MiniHeader

Reply:
Can we add a sanity check that the logs are correctly encoded/decoded as well?

Reply:
Aren't we checking that at line 676 where we compare that the inserted value equals the retrieved value from the DB? Or are we talking about different things?

Reply:
I wasn't sure if `testify/assert` handled the equality comparison for a nested array of structs properly, but after trying it out locally, it doesn't seem to be a problem. Never mind!